app/src/ui/lib/augmented-filter-list.tsx

```diff
@@ -237,26 +253,22 @@ export class AugmentedSectionFilterList<
     prevState: IAugmentedSectionFilterListState<T>
   ) {
     if (this.props.onSelectionChanged) {
-      const oldSelectedItemId = getItemIdFromRowIndex(
-        prevState.rows,
-        prevState.selectedRow
+      const oldSelectedItemIds = prevState.selectedRows.map(row =>
+        getItemIdFromRowIndex(prevState.rows, row)
       )
-      const newSelectedItemId = getItemIdFromRowIndex(
-        this.state.rows,
-        this.state.selectedRow
+      const newSelectedItemIds = this.state.selectedRows.map(row =>
+        getItemIdFromRowIndex(this.state.rows, row)
       )
 
-      if (oldSelectedItemId !== newSelectedItemId) {
-        const propSelectionId = this.props.selectedItem
-          ? this.props.selectedItem.id
-          : null
-
-        if (propSelectionId !== newSelectedItemId) {
-          const newSelectedItem = getItemFromRowIndex(
-            this.state.rows,
-            this.state.selectedRow
-          )
-          this.props.onSelectionChanged(newSelectedItem, {
+      if (xor(oldSelectedItemIds, newSelectedItemIds).length > 0) {
```

Comment:
Can we add a bit of documentation here for future readers of this code. xor isn't a common operation to see so I think it warrants explaining what we're trying to do.